src/compiler/cell_manager.rs

```diff
@@ -249,6 +334,11 @@ impl CellManager for MaxWidthCellManager {
             placement.steps.insert(Rc::clone(step), step_placement);
         }
 
+        // Redundant if we do the step height check at the front.
+        // if !placement.same_height() {
+        //     panic!("To use shared signal, all steps placed by MaxWidthCellManager need to have
+        // the same height. Please use a different cell manager."); }
```

Comment:
Let me know if you want me to do this check at the front (so that I delete these comments) or at the back (so that I uncomment these).